authzed/api/v1/permission_service.proto

```diff
@@ -250,9 +250,39 @@ message DeleteRelationshipsRequest {
   repeated Precondition optional_preconditions = 2
       [ (validate.rules).repeated .items.message.required =
             true ]; // To be bounded by configuration
+
+  // optional_limit, if non-zero, specifies the limit on the number of relationships to be deleted.
+  // If there are more matching relationships found to be deleted than the limit specified here,
+  // the deletion call will fail with an error to prevent partial deletion. If partial deletion
+  // is needed, specify below that partial deletion is allowed. Partial deletions can be used
+  // in a loop to delete large amounts of relationships in a *non-transactional* manner.
+  uint32 optional_limit = 3 [(validate.rules).uint32 = {gte:0, lte: 1000}];
```

Comment:
Same as with Cursor limits: I think this should be left to the server to be defined and validated in runtime, instead of defining it at the API level.

Reply:
Perhaps in V2, but in order to make this fully back compat, we'll make it client specified for now and offer suggestions